### PR TITLE
Add a canary function; role example

### PR DESCRIPTION
**Pull request by tptee**
- Adds an example canary function using the new submodule in https://github.com/FormidableLabs/terraform-aws-serverless/pull/52
- Adds an example of creating an assumed role from the new policy outputs in https://github.com/FormidableLabs/terraform-aws-serverless/pull/52. Comments detail its use in cross-account delegation and ease of testing.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -8,6 +8,8 @@ terraform {
   }
 }
 
+data "aws_caller_identity" "current" {}
```
Comment (ryan-roemer) [inline terraform/main.tf]: Let's move this to the file in which it's actually used...

File: terraform/role-ci.tf [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Example of how to use policies from terraform-aws-serverless to create
```
Comment (ryan-roemer) [inline terraform/role-ci.tf:1]: How far off are we placing something like this in `canary` submodule or even `canary-role-ci` submodule (for after `canary`) in terraform-aws-serverless?

Comment (tptee) [reply terraform/role-ci.tf:1]: There's a few things I'd need to figure out before introducing this to `terraform-aws-serverless`:
- can we automatically compose `assume_role_policy`s from multiple modules? Is it enough to put any role principals (e.g. `lambda.amazonaws.com`) in the main module instead, even if their relevant submodule isn't used?
- if not automatic, does it make sense to export the principals from each module and have the user combine them?

Comment (tptee) [reply terraform/role-ci.tf:1]: Ticketed in https://github.com/FormidableLabs/terraform-aws-serverless/issues/53

**Comment by tptee** [comment]
If that ticket's enough for now, this is ready for another look!

**Comment by ryan-roemer** [comment]
Apologies -- My refactor created conflicts. Mind resolving those and refactoring to normalize `# OPTION(name): COMMENT` style comments in SLS + TF configs? Thanks!!!

**Comment by tptee** [comment]
No worries, on it!

File: terraform/role-ci.tf [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
```
Comment (ryan-roemer) [inline terraform/role-ci.tf:1]: Let's add a comment at the top here with link to FormidableLabs/terraform-aws-serverless#53 stating we're looking to move this into the module...

**Review by ryan-roemer — APPROVED** [review]
One minor comment addition, rest LGTM. Nice work! 🎉

**Pull request merged**